config/src/config.rs

```diff
@@ -102,80 +106,94 @@ impl Default for BaseConfig {
     fn default() -> BaseConfig {
         BaseConfig {
             data_dir_path: PathBuf::from("."),
+            role: RoleType::Validator,
             temp_data_dir: None,
         }
     }
 }
 
 impl BaseConfig {
-    /// Constructs a new BaseConfig with an empty temp directory
-    pub fn new(data_dir_path: PathBuf) -> Self {
+    pub fn new(data_dir_path: PathBuf, role: RoleType) -> Self {
         BaseConfig {
             data_dir_path,
+            role,
             temp_data_dir: None,
         }
     }
 }
 
-#[cfg(any(test, feature = "fuzzing"))]
 impl Clone for BaseConfig {
     fn clone(&self) -> Self {
         Self {
             data_dir_path: self.data_dir_path.clone(),
+            role: self.role,
             temp_data_dir: None,
         }
     }
 }
 
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum RoleType {
+    Validator,
+    FullNode,
+}
+
+impl RoleType {
+    pub fn is_validator(&self) -> bool {
+        *self == RoleType::Validator
```

Comment:
I see that you added this but then just didn't use it?

Reply:
https://github.com/libra/libra/blob/master/config/src/config/mod.rs#L195
https://github.com/libra/libra/blob/master/config/src/config/network_config.rs#L131

I think there was at least one or two more use cases but they probably got lost in rebase hell. Might have been better to separate certain concerns into commits to be appended so as to not have such circumstances.